cupy/cudnn.pyx

```diff
@@ -77,6 +77,8 @@ cpdef _create_tensor_nd_descriptor(
         size_t desc, core.ndarray arr, int data_type):
     cdef vector.vector[int] c_shape, c_strides
     cdef Py_ssize_t itemsize, s
+    if data_type == -1:
+        data_type = get_data_type(arr.dtype)
```

Comment:
What does `data_type == -1` mean?
In general, more comments (function comments/code comments) would be appreciated, so that review (and improvements) would be easier.